### ISSUE TITLE
"Assertion Failed: Cannot update watchers for `loadedText` on `<my-app@component:infinity-loader::ember1471>` after it has been destroyed."

### ISSUE THREAD

**Issue by alexmiddeleer** [hhff/ember-infinity#248]
Seeing this error with `destroyOnInfinity=true`. Ember 2.18. and ember infinity 0.2.8. This is just after upgrading ember. 

```
vendor.js:27629 Uncaught EmberError {stack: "Error: Assertion Failed: Cannot update watchers fo…(http://localhost:4200/assets/vendor.js:22013:31)", description: undefined, fileName: undefined, lineNumber: undefined, message: "Assertion Failed: Cannot update watchers for `load…-loader::ember1471>` after it has been destroyed.", …}code: undefineddescription: undefinedfileName: undefinedlineNumber: undefinedmessage: "Assertion Failed: Cannot update watchers for `loadedText` on `<rallypoint-mobile-fe@component:infinity-loader::ember1471>` after it has been destroyed."name: "Error"number: undefinedstack: "Error: Assertion Failed: Cannot update watchers for `loadedText` on `<rallypoint-mobile-fe@component:infinity-loader::ember1471>` after it has been destroyed.↵    at new EmberError (http://localhost:4200/assets/vendor.js:27386:25)↵    at Object.assert (http://localhost:4200/assets/vendor.js:27629:15)↵    at Meta.writeWatching (http://localhost:4200/assets/vendor.js:38135:54)↵    at watchKey (http://localhost:4200/assets/vendor.js:37274:13)↵    at new RootPropertyReference (http://localhost:4200/assets/vendor.js:35135:42)↵    at RootReference.get (http://localhost:4200/assets/vendor.js:35052:52)↵    at http://localhost:4200/assets/vendor.js:15119:28↵    at AppendOpcodes.evaluate (http://localhost:4200/assets/vendor.js:14851:13)↵    at VM.next (http://localhost:4200/assets/vendor.js:22027:32)↵    at VM.execute (http://localhost:4200/assets/vendor.js:22013:31)"__proto__: ExtendableBuiltinconstructor: ƒ EmberError(message)__proto__: Errorconstructor: ƒ ExtendableBuiltin()__proto__: constructor: ƒ Error()message: ""name: "Error"toString: ƒ toString()__proto__: Object
assert @ vendor.js:27629
writeWatching @ vendor.js:38135
watchKey @ vendor.js:37274
RootPropertyReference @ vendor.js:35135
get @ vendor.js:35052
(anonymous) @ vendor.js:15119
evaluate @ vendor.js:14851
next @ vendor.js:22027
execute @ vendor.js:22013
handleException @ vendor.js:21412
handleException @ vendor.js:21605
_throw @ vendor.js:21291
evaluate @ vendor.js:15715
execute @ vendor.js:21278
rerender @ vendor.js:21636
_this.render @ vendor.js:33146
runInTransaction @ vendor.js:36561
_renderRoots @ vendor.js:33374
_renderRootsTransaction @ vendor.js:33404
_revalidate @ vendor.js:33441
invoke @ vendor.js:23174
flush @ vendor.js:23067
flush @ vendor.js:23245
end @ vendor.js:23378
_run @ vendor.js:23715
_join @ vendor.js:23691
join @ vendor.js:23445
loopEnd @ vendor.js:33222

``

**Comment by alexmiddeleer** [comment]
While removing `destroyOnInfinity=true` removed the error, it revealed another; the loader template is behaving as if `infinityModel.reachedInfinity` is false.

**Comment by snewcomer** [comment]
@alexmiddeleer Thanks for the issue!  I would try one thing first:


1.  We are looking to release a 1-0 [branch](https://github.com/hhff/ember-infinity/tree/1-0) in the coming weeks.  If you add this line to your package.json -> `"ember-infinity": "git+ssh://git@github.com:hhff/ember-infinity.git#1-0"` and let me know if that works for you.  Note, the property `destroyOnInfinity` was changed to `hideOnInfinity` [here](https://github.com/hhff/ember-infinity/pull/244).

**Comment by alexmiddeleer** [comment]
@snewcomer Thanks, that fixed the issue for me! The issue with reaching infinity I made in my second comment turned out to be a pre-existing problem that we have been ignoring for years probably.

**Comment by alexmiddeleer** [comment]
@snewcomer Looks like there was a regression with the infinityModelUpdated event in that revision - the event doesn't seem to fire, at least for the initial records. We are just going to use css as a workaround and stick to version 2.8 for a bit. I'm sorry for being vague about the regression, I will try to follow up with more details if I find any / have time.

**Comment by snewcomer** [comment]
@alexmiddeleer Happy to help out.  See the line below for the that part of the lib and lmk what you see.  Also, 1-0 was just merged so you can install `1.0.0-alpha.4` to try it out.

https://github.com/ember-infinity/ember-infinity/blob/master/addon/mixins/route.js#L309
